src/Data/Geometry/Geos/Geometry.hs

```diff
@@ -125,8 +127,8 @@ convertLineStringToRaw (LineString cs) s = do
 
 convertPolygonToRaw :: R.Geometry a => Polygon -> SRID -> Geos a
 convertPolygonToRaw (Polygon lrs) s = do
-  ext <- convertLinearRingToRaw (V.head lrs) s
-  inn <- (flip convertLinearRingToRaw $ s) `V.mapM` V.tail lrs
+  ext :: R.GeomConst <- convertLinearRingToRaw (V.head lrs) s
+  inn :: V.Vector R.GeomConst <- (flip convertLinearRingToRaw $ s) `V.mapM` V.tail lrs
   R.createPolygon ext (V.toList inn)  >>= R.setSRID s
 
 convertMultiPointToRaw :: R.Geometry a => MultiPoint -> SRID -> Geos a
```

Comment:
That's right (though it's sufficient to ensure that the R.create* functions take a GeomConst so I'm removing the scoped type variables here.
Interestingly, I was unable to reproduce the same memory bug with the MultiPoint, MultiLineString, (etc) functions.